### PR TITLE
Added course list page.

### PR DESCRIPTION
**Pull request by waheedahmed**
[ECOM-6085](https://openedx.atlassian.net/browse/ECOM-6085)
[ECOM-6086](https://openedx.atlassian.net/browse/ECOM-6086)
[ECOM-6448](https://openedx.atlassian.net/browse/ECOM-6448)

Created course listing page.

**Screenshots:**
screencapture-localhost-8004-publisher-courses-1482140467049

screencapture-localhost-8004-publisher-courses-1482140431457

### CODE REVIEW

File: course_discovery/apps/publisher/views.py [hunk 1]
```diff
@@ -412,3 +412,19 @@ def post(self, request):
 class UpdateCourseKeyView(mixins.LoginRequiredMixin, mixins.ViewPermissionMixin, UpdateAPIView):
     queryset = CourseRun.objects.all()
     serializer_class = UpdateCourseKeySerializer
+
+
+class CourseListView(mixins.LoginRequiredMixin, ListView):
```
Comment (awais786) [inline course_discovery/apps/publisher/views.py:417]: Make this as a mix-in.

**Review by awais786 — APPROVED** [review]

File: course_discovery/apps/publisher/views.py (hunk 1 above)
Comment (waheedahmed) [reply course_discovery/apps/publisher/views.py:417]: @awais786 Both ```get_querysets()``` are different, this is for courses and the other one in Dashboard is for course runs.

**Comment by waheedahmed** [comment]
@bderusha Please review this PR.

File: course_discovery/apps/publisher/views.py [hunk 2]
```diff
@@ -412,3 +412,19 @@ def post(self, request):
...
+    template_name = 'publisher/courses.html'
+
+    def get_queryset(self):
+        user = self.request.user
+        if is_publisher_admin(user):
+            courses = Course.objects.all()
+        elif is_internal_user(user):
+            courses = Course.objects.filter(course_user_roles__user=user).distinct()
+        else:
+            courses = get_objects_for_user(user, Course.VIEW_PERMISSION, Course)
```
Comment (bderusha) [inline course_discovery/apps/publisher/views.py:428]: Shouldn't we be using the `OrganizationExtension` for permissions?

**Review by bderusha — COMMENTED** [review]
This looks good to me. Only one question about how we are doing permissions.

File: course_discovery/apps/publisher/views.py (hunk 2 above)
Comment (awais786) [reply course_discovery/apps/publisher/views.py:428]: @bderusha I am working 6628 that will add the new permissions and then we will update the code in publisher app.

Comment (waheedahmed) [reply course_discovery/apps/publisher/views.py:428]: @bderusha Yes, we will use ```OrganizationExtension``` but right now ```VIEW_PERMISSION``` is not available in it. @awais786 is working on that and we will update this in that PR.

**Review by bderusha — APPROVED** [review]

**Pull request merged**